botorch/models/gpytorch.py

```diff
@@ -489,7 +496,7 @@ def condition_on_observations(
         fantasy_model._input_batch_shape = fantasy_model.train_targets.shape[
             : (-1 if self._num_outputs == 1 else -2)
         ]
-        fantasy_model._aug_batch_shape = fantasy_model.train_targets.shape[:-1]
+
```

Comment:
I know that this is not really used outside of model initialization, but it'd still be good to keep it consistent with the model. Let's set it conditional on the model type.
```suggestion
        if not self._is_fully_bayesian:
            fantasy_model._aug_batch_shape = fantasy_model.train_targets.shape[:-1]
```